WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazePaymentMethodsView.swift

```diff
@@ -163,21 +140,19 @@ struct BlazePaymentMethodsView: View {
         if let addPaymentMethodURL = viewModel.addPaymentMethodURL,
            let fetchPaymentMethodURLPath = viewModel.addPaymentSuccessURL {
             NavigationView {
-                AuthenticatedWebView(isPresented: $showingAddPaymentWebView,
+                AuthenticatedWebView(isPresented: $viewModel.showingAddPaymentWebView,
                                      url: addPaymentMethodURL,
-                                     urlToTriggerExit: fetchPaymentMethodURLPath) {
-                    showingAddPaymentWebView = false
-                    Task {
-                        await viewModel.syncPaymentInfo()
-                    }
+                                     urlToTriggerExit: fetchPaymentMethodURLPath) { url in
+                    viewModel.showingAddPaymentWebView = false
                     viewModel.notice = Notice(title: Localization.paymentMethodAddedNotice, feedbackType: .success)
+                    viewModel.didAddNewPaymentMethod(successURL: url)
                 }
                                      .navigationTitle(Localization.paymentMethodWebViewTitle)
                                      .navigationBarTitleDisplayMode(.inline)
                                      .toolbar {
                                          ToolbarItem(placement: .confirmationAction) {
                                              Button(action: {
-                                                 showingAddPaymentWebView = false
+                                                 viewModel.showingAddPaymentWebView = false
                                              }, label: {
                                                  Text(Localization.doneButtonAddPayment)
                                              })
```

Comment:
Nit: since the screen is now dismissed automatically after successfully adding a payment method, should we change the button to Cancel instead? 

The design uses a push navigation, but I think using modal is good too.

<img width="277" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/00d8ffc3-c7ee-4ab2-8d07-ffbdaed0b2d0">

Reply:
I started using "Cancel" button in https://github.com/woocommerce/woocommerce-ios/pull/11754/commits/e2de5e0a7f984bcb20e3111a5834832d01acb148